default-challenge-bypass-ristretto-ffi.nix

```diff
@@ -0,0 +1,18 @@
+# Provide a no-argument function for building the underlying
+# challenge-bypass-ristretto ffi bindings.
+#
+# This defaults to pinning a pretty new version of nixpkgs because from around
+# 19.09 and older the nixpkgs Rust build system does something weird with the
+# library output and I can't figure out how to make packaging consistent
+# between 19.09-and-older and newer-than-19.09.
+#
+# Supply your own nixpkgs if you want, as long as it's newer than 19.09.
+let
+  nixpkgs = builtins.fetchTarball {
+    name = "nixpkgs";
+    url = "https://github.com/NixOS/nixpkgs/archive/refs/tags/21.05.tar.gz";
```

Comment:
Is this really a stable URL? Since 21.05 is a moving target - less than unstable (hopefully!) but still...

Reply:
Since it's a tag, I think so?  I know you *can* move git tags but that's really rude, right?  It looks like 21.05 is set to a commit with message "Release 21.05" from May 31st.  I think that's a pretty good hint that nixos is not moving this tag around.

There is a branch named `release-21.05` which has received commits as recently as July 12th.  I think that's probably the moving target you were thinking of.